Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -73,425 +75,388 @@
+    public ResponseEntity<String> addConceptCollectionsToWorkspace(@PathVariable("workspaceId") String workspaceId,
+            HttpServletRequest request, HttpServletResponse response, @RequestBody String xml,
+            @RequestHeader("Accept") String accept, ModelMap model, Principal principal)
+                    throws RestException, QuadrigaStorageException, QuadrigaAccessException {
+        IUser user = usermanager.getUser(principal.getName());
+        String ccName = request.getParameter("name");
+        String desc = request.getParameter("desc");
+
+        if (!checkWSSecurity.checkIsWorkspaceExists(workspaceId)) {
+            String errorMsg = restMessage.getErrorMsg("Workspace ID : " + workspaceId + " doesn't exist", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+        }
+
+        IConceptCollection collection = conceptCollectionFactory.createConceptCollectionObject();
+
+        if (ccName == null || ccName.isEmpty()) {
+            String errorMsg = restMessage.getErrorMsg("Please provide concept collection name", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+        }
+        if (desc == null || desc.isEmpty()) {
+            String errorMsg = restMessage.getErrorMsg("Please provide concept collection description", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+        }
+        logger.debug("XML : " + xml);
+        JAXBElement<QuadrigaConceptReply> response1 = null;
+        try {
+            JAXBContext context = JAXBContext.newInstance(QuadrigaConceptReply.class);
+            Unmarshaller unmarshaller = context.createUnmarshaller();
+            unmarshaller.setEventHandler(new javax.xml.bind.helpers.DefaultValidationEventHandler());
+            InputStream is = new ByteArrayInputStream(xml.getBytes());
+            response1 = unmarshaller.unmarshal(new StreamSource(is), QuadrigaConceptReply.class);
+        } catch (JAXBException e) {
+            logger.error("Error in unmarshalling", e);
+            String errorMsg = restMessage.getErrorMsg("Error in unmarshalling", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+        }
+
+        QuadrigaConceptReply qReply = response1.getValue();
+        ConceptList conList = qReply.getConceptList();
+        List<Concept> conceptList = conList.getConcepts();
+        if (conceptList.size() < 1) {
+            String errorMsg = restMessage.getErrorMsg("Concepts XML is not valid", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+        }
+
+        collection.setDescription(desc);
+        collection.setOwner(user);
+        collection.setConceptCollectionName(ccName);
+
+        conceptControllerManager.addConceptCollection(collection);
+        String ccId = conceptControllerManager.getConceptCollectionId(ccName);

```

Comment:
this is not good. can you do the following:
in ConceptCollectionDAO in addCollection -> set the id of the conceptCollection that is given to the method after you saved the collection (and therefore have an id).
then use the getId method of the conceptcollection in this line to get the id (rather than querying by ccname, CCname doesn't have to be unique, so this might result in errors).